app/org/sagebionetworks/bridge/play/controllers/StudyController.java

```diff
@@ -190,8 +193,42 @@ public Result getEmailStatus() throws Exception {
         return okResult(new EmailVerificationStatusHolder(status));
     }
 
+    /** Resends the verification email for the current study's email. */
     @BodyParser.Of(BodyParser.Empty.class)
-    public Result verifyEmail() throws Exception {
+    public Result resendVerifyEmail(String type) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        StudyEmailType parsedType = parseEmailType(type);
+        studyService.sendVerifyEmail(session.getStudyIdentifier(), parsedType);
+        return okResult("Resending verification email for consent notification email.");
+    }
+
+    /**
+     * Verifies the emails for the study. Since this comes in from an email with a token, you don't need to be
+     * authenticated. The token itself knows what study this is for.
+     */
+    @BodyParser.Of(BodyParser.Empty.class)
+    public Result verifyEmail(String identifier, String token, String type) {
+        StudyEmailType parsedType = parseEmailType(type);
+        studyService.verifyEmail(new StudyIdentifierImpl(identifier), token, parsedType);
+        return okResult("Consent notification email address verified.");
+    }
+
+    // Helper method to parse and validate the email type for study email verification workflow. We do verification
+    // here so that the service can just deal with a clean enum.
+    private static StudyEmailType parseEmailType(String typeStr) {
+        if (StringUtils.isBlank(typeStr)) {
+            throw new BadRequestException("Email type must be specified");
+        }
+
+        try {
+            return StudyEmailType.valueOf(typeStr);
```

Comment:
Because all the enums exposed in the API are lowercase, I would use typeStr.toUpperCase() here

Reply:
Changed.